File: classes/map/Tile.py
from random import randint, choice
from classes.map.Resource import Resource

class Tile:
	def __init__(self):
		self.resources = []
		self.__populateResources()

	def __populateResources(self):
		for index in range(randint(2,6)):
			self.resources.append(Resource(choice(['food', 'water', 'sex'])))

File: Assets/Classes/Map/Tile.cs
using System;

public class Tile {
	public Resource[] resources;
	private Random rand;

	public Tile() {
		rand = new Random ();
		resources = new Resource[rand.Next(2,6)];
		populateResources ();
	}

	private void populateResources() {
		for (int i = 0; i < resources.Length; i++) {
			string resource = Resource.RESOURCE_TYPES [rand.Next (Resource.RESOURCE_TYPES.Length)];
			resources [i] = new Resource(resource);
		}
	}
}
